src/components/products-list.vue

```diff
@@ -0,0 +1,79 @@
+<template> 
+<div :class="'product'+(index+1)" v-for="product, index in products"> 
+      <img :src="product.img" imgfield="img" style="max-width: 150px; width: 150px;" alt=""> 
+      <p>{{ product.name }}</p> 
+    </div>
+</template>
+<script setup lang="ts">
+
+type Product =  {
```

Comment:
Move type Product to stores/product.ts

import it to product page and product-list 
`import type { Product } from '@/stores/product`